### PR TITLE
Setup mergify 

### PR DESCRIPTION
**Pull request by vigoux**
As discussed internally, we setup mergify.

### CODE REVIEW

File: .github/mergify.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+pull_request_rules:
+  - name: Automatic merge on approval
+    conditions:
+      - and:
+        - "-draft"
+        - "#check-failure=0"
+        - "#approved-reviews-by>=1"
```
Comment (theHamsta) [inline .github/mergify.yml:7]: I don't know whether we want to auto-merge on the first approval. At the moment I merge trivial things right away will I leave other things I approve open for a while to give other the chance to give a look.

**Review by theHamsta — APPROVED** [review]

**Comment by stsewd** [comment]
So, is the usage like this?

```
@Mergifyio backport 5.0
```

**Comment by stsewd** [comment]
was hoping the bot would ignore that :D

File: .github/mergify.yml (hunk 1 above)
Comment (nifoc) [reply .github/mergify.yml:7]: This did indeed cause #1477 to be merged by accident

**Pull request merged**